integration/library_test.go

```diff
@@ -0,0 +1,121 @@
+// +build integration
+
+package integration
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/aquasecurity/fanal/analyzer"
+	"github.com/aquasecurity/fanal/extractor/docker"
+	"github.com/stretchr/testify/require"
+
+	_ "github.com/aquasecurity/fanal/analyzer/command/apk"
+	_ "github.com/aquasecurity/fanal/analyzer/library/bundler"
+	_ "github.com/aquasecurity/fanal/analyzer/library/cargo"
+	_ "github.com/aquasecurity/fanal/analyzer/library/composer"
+	_ "github.com/aquasecurity/fanal/analyzer/library/npm"
+	_ "github.com/aquasecurity/fanal/analyzer/library/pipenv"
+	_ "github.com/aquasecurity/fanal/analyzer/library/poetry"
+	_ "github.com/aquasecurity/fanal/analyzer/library/yarn"
+	_ "github.com/aquasecurity/fanal/analyzer/os/alpine"
+	_ "github.com/aquasecurity/fanal/analyzer/os/amazonlinux"
+	_ "github.com/aquasecurity/fanal/analyzer/os/debianbase"
+	_ "github.com/aquasecurity/fanal/analyzer/os/photon"
+	_ "github.com/aquasecurity/fanal/analyzer/os/redhatbase"
+	_ "github.com/aquasecurity/fanal/analyzer/os/suse"
+	_ "github.com/aquasecurity/fanal/analyzer/pkg/apk"
+	_ "github.com/aquasecurity/fanal/analyzer/pkg/dpkg"
+	_ "github.com/aquasecurity/fanal/analyzer/pkg/rpm"
+
+	"github.com/aquasecurity/fanal/cache"
+	"github.com/aquasecurity/fanal/types"
+)
+
+func TestFanal_Library_DockerMode(t *testing.T) {
+	ctx := context.Background()
```

Comment:
First of all, we should do a table driven test because we need to test all OSes.

Reply:
Yeah I had kept this open for discussion. I wasn't sure how much detail we wanted to go into with it. Shall we add a test for each OS? Or one for each OS and the variety of options like here https://github.com/aquasecurity/trivy/blob/master/integration/standalone_test.go#L31-L343?

Reply:
Let's add a test for each OS. We don't have many options for fanal like Trivy.